### ISSUE TITLE
Some problem about xgboost model.

### ISSUE THREAD

**Issue by Echo9573** [sql-machine-learning/sqlflow#1411]
There are some problems with xgboost model in sqlflow: 
the following train and validation set and the parameter are the same on both LOCAL & SQLFlow .
`BUG`：

**1. Incorrect predict result in xgboost model**, which is shown as follows:
![image](https://user-images.githubusercontent.com/24820352/70767044-b827b300-1d9a-11ea-8f8b-1ce51b6fa428.png)
![image](https://user-images.githubusercontent.com/24820352/70767147-0937a700-1d9b-11ea-9052-d2c6aa4587e9.png)

**2.Incorrect evaluation results when training xgboost model**
-  LOCAL
![image](https://user-images.githubusercontent.com/24820352/70767322-d6da7980-1d9b-11ea-9ab2-64aaa339b719.png)

-  SQLFlow
![image](https://user-images.githubusercontent.com/24820352/70767346-ea85e000-1d9b-11ea-98ac-8df95b8b5d33.png)

`To Explore`:

**3. Inconsistent with local iteration effect.**
-  LOCAL
![image](https://user-images.githubusercontent.com/24820352/70767238-69c6e400-1d9b-11ea-9d9d-13b1b16582db.png)

-  SQLFlow
![image](https://user-images.githubusercontent.com/24820352/70767220-53b92380-1d9b-11ea-82fc-f222205fc197.png)

`Complete information`
The Complete information files please download this zip file:
[compare_local_SQLFlow_xgboost.zip](https://github.com/sql-machine-learning/sqlflow/files/3958871/compare_local_SQLFlow_xgboost.zip)

**Comment by shendiaomo** [comment]
Do we have any progress? @weiguoz @Yancey1989

**Comment by Yancey0623** [comment]
Hi @shendiaomo , this case looks like #1397 and @tonyyang-svail follows up it. I will also follow up on this case today.
